src/osd/PrimaryLogPG.h

```diff
@@ -1622,18 +1622,25 @@ class PrimaryLogPG : public PG, public PGBackend::Listener {
       > reactions;
     struct ReservationCB : public Context {
       PrimaryLogPGRef pg;
+      OSDService *service;
       bool canceled;
-      ReservationCB(PrimaryLogPG *pg) : pg(pg), canceled(false) {}
+      ReservationCB(PrimaryLogPG *pg) : pg(pg), service(pg->osd),
+					canceled(false) {}
       void finish(int) override {
-	pg->lock();
+	PrimaryLogPG *pgp = pg.get();
+	if (service->is_stopping() || !pgp)
```

Comment:
ahh, that explains it! i need to look at this part closer before sending out more noise.  sorry and thanks !